controller/workitem.go

```diff
@@ -524,6 +526,25 @@ func ConvertJSONAPIToWorkItem(ctx context.Context, method string, appl applicati
 	return nil
 }
 
+// replaceFieldNames modifies the source to rename all system.* fields to
+// system_* fields. It uses the OldToNewFieldNameMap to find the correct new
+// name for the old field
+// TODO(ibrahim) - Remove these after fields are migrated
+func replaceFieldNames(source *app.WorkItem) {
+	attributeList := source.Attributes
+	for oldAttributeName := range attributeList {
+		newAttributeName, ok := workitem.OldToNewFieldNameMap[oldAttributeName]
+		if ok {
+			// Assign system.* value to system_* only if system_* doesn't exist
+			// system_* has precedence over system.*
+			if _, empty := source.Attributes[newAttributeName]; !empty {
+				source.Attributes[newAttributeName] = source.Attributes[oldAttributeName]
+			}
+			delete(source.Attributes, oldAttributeName)
```

Comment:
Is this the reasons for why some fields are replaced in the golden files? Because I could see that others are just added and not removed.

Reply:
NO no. The replaced fields are in the `relationships`. We cannot add two relationships for the same attribute because the `workitemRelationships` is defined as
```go
// WorkItemRelationships defines only `assignee` as of now. To be updated
var workItemRelationships = a.Type("WorkItemRelationships", func() {
	a.Attribute("assignees", relationGenericList, "This defines assignees of the Work Item")
	a.Attribute("labels", relationGenericList, "List of labels attached to the Work Item")
	a.Attribute("system_boardcolumns", relationGenericList, "List of board columns this Work Item is attached to")
	a.Attribute("creator", relationGeneric, "This defines creator of the Work Item")
	a.Attribute("baseType", relationBaseType, "This defines type of Work Item")
	a.Attribute("comments", relationGeneric, "This defines comments on the Work Item")
	a.Attribute("iteration", relationGeneric, "This defines the iteration this work item belong to")
	a.Attribute("area", relationGeneric, "This defines the area this work item belongs to")
	a.Attribute("children", relationGeneric, "This defines the children of this work item")
	a.Attribute("space", relationSpaces, "This defines the owning space of this work item.")
	a.Attribute("parent", relationKindUUID, "This defines the parent of this work item.")
	a.Attribute("workItemLinks", relationGeneric, "List of links in which this work item is involved")
	a.Attribute("events", relationGeneric, "List of events in which this work item is involved")
})
```